### PR TITLE
chore(mars-theme): add 404 page

### PR DESCRIPTION
**Pull request by DAreRodz**

### CODE REVIEW

File: packages/mars-theme/src/html/components/theme.js [hunk 1]
```diff
@@ -30,6 +31,7 @@ const Theme = ({ state }) => (
     <Body>
       {state.source.data(state.router.path).isArchive && <List />}
       {state.source.data(state.router.path).isPostType && <Post />}
+      {state.source.data(state.router.path).is404 && <Page404 />}
```
Comment (orballo) [inline packages/mars-theme/src/html/components/theme.js:34]: I tried this yesterday and `is404` was undefined when I entered a wrong url. When exactly is `is404` populated?

I think it needs some refactoring in `wp-source` in order to cover all the cases where there is no content available.

Comment (DAreRodz) [reply packages/mars-theme/src/html/components/theme.js:34]: @orballo, what URL did you try? The property `is404` is set [here](https://github.com/frontity/frontity/blob/dev/packages/wp-source/src/actions.ts#L35), when a handler throws an exception.

Comment (orballo) [reply packages/mars-theme/src/html/components/theme.js:34]: I tried something invented like `/invented-path/` or something like that. Maybe the error was somewhere else and I was mistaken?

**Review by luisherranz — APPROVED** [review]
It works fine in my local environment :)

**Review by orballo — APPROVED** [review]

File: packages/mars-theme/src/html/components/theme.js (hunk 1 above)
Comment (DAreRodz) [reply packages/mars-theme/src/html/components/theme.js:34]: I don't know, but it seems to be working fine now.

**Pull request merged**